New indev map: Nadir Extraction Site

Kubius opened the pull request:
<!-- The text between the arrows are comments - they will not be visible on your PR. -->
<!-- To automatically tag this PR, add the uppercase label(s) surrounded by brackets below, for example: [LABEL] -->
[FEATURE]
## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->

![nadir](https://user-images.githubusercontent.com/12454065/157130511-e69d906e-0813-4926-b7d1-1aaeba6d4256.png)

Adds a new map, the Nadir Extraction Site, and associated features including three proprietary "main" systems. **Does not introduce the map to rotation.**

The map is operational and has been received some small-scale multiplayer tests with a personal server, but I'm to the point where I'm going to need considerable assistance getting this from operational to polished (described later on), so it's PR time!

### Lore Blurb (Headline/Contents)

> Nadir Extraction Site - Historical Entry: "Bitter Terminus"
> 
> Deep within a pit of caustic acid on the frigid and storm-wracked frontier world of Magus, the Nadir Extraction Site stands as a testament to the ambition and avarice of Nanotrasen's finest; constructed to extract and transmute a vast deposit of the anomalous resource Miraclium, it serves both as an effective mine and a unique opportunity for experimentation. Surrounded by a flesh-melting sea and connected to the outside world only by a prototype transception antenna, purpose-built to pierce the corrosive sleet that ravages the planetary surface, Nadir's brutal isolation and ever-looming danger have earned it a reputation among Nanotrasen employees as the "point of no return."

### Novel Station Features

Map-installed hardware newly created for Nadir. **All of these are operational**, though further tuning and upgrades should be involved.

**Harmonic Siphon**

![ndcore](https://user-images.githubusercontent.com/12454065/157130828-53a66fc7-c197-4353-ad7d-6981e2507209.png)

The Harmonic Siphon is Nadir's centerpiece; situated directly over the extraction site's Miraclium deposit, Scientists and Engineers work in conjunction to configure and utilize it for material harvesting. Nadir does not have a dedicated Miner role, as the surrounding acid poses an ongoing hazard that makes conventional mining dangerous.

Siphon configuration takes place using devices called resonators, which serve multiple functions in the extraction process. Each resonator influences up to three extraction parameters (lateral resonance, vertical resonance and shear) based on their position and calibrated intensity; the sum of active resonators' intensity influences both the power draw of the setup, and the rate at which resources are extracted and transmuted. There are two types of resonator that can be manufactured, though only one type is required for most operation (four of said type are included at round start).

Configurations for these materials vary in requirements and difficulty. Basic materials like char, mauxite and pharosium come with reference configurations, so those unfamiliar with the systems are able to extract necessary materials for baseline operation; some sophisticated materials like starstone have randomized requirements and considerably higher power draw.

The siphon is affected by erroneous configurations; excessive shear values, panels left open or too much cumulative intensity for the targeted product will cause resonators to fail, knocking them out of the set and requiring them to be rewired. Exceedingly high shear values can also result in explosive resonator failure or anomalous behavior.

**Catalytic Generators**

![ndcata](https://user-images.githubusercontent.com/12454065/157130869-38b1707e-6a5d-42fc-9c3d-b0170246000b.png)

Extraction site operation is driven by a pair of catalytic substations, located at the north and south tips of the station. Each substation is equipped with two catalytic generators, which insert a pair of catalytic rods into the acid ocean, utilizing a quirk in its composition to generate electricity.

Catalytic rods can be manufactured in station nano-fabricators; the desired material properties for anode and cathode style rods differ significantly, with exceedingly effective rods yielding an exponential increase in power output. This is tempered somewhat by the generators' operation; the lowest maximum power output between the two rods is the one which is actually used.

Rods have an "infinite" lifespan, but degrade gradually in performance over time based on their corrosion resistance; degradation is a continual multiplication, so the rate of decrease slows over time.

The station's power setup has been provisioned fairly generously - each generator has one paired SMES set to 50000/25000, and the SMES continue charging until about 28 minutes into the shift (figure attained in local live testing), sustaining just under an hour of baseline power. Basic rod replacement is as simple as pharosium and mauxite.

**Transception Array**

![ndarray](https://user-images.githubusercontent.com/12454065/157130987-13c10ca3-0d74-42e8-a709-cd2db8461515.png)

As most objects without active shielding (i.e. crates) would likely have a dubious survival rate when passing through active sleetstorms, an alternative method for transport had to be devised; the Transception Array serves this role, as well as the role of a conventional communications dish.

Interfacing with this array's cargo transport capability is done using specialized cargo pads, two of which are included in the cargo bay's transception endpoint; they can be scanned and reproduced by Mechanics to create cargo shipping and receiving capability anywhere on the station, so long as there is a connection to the same powernet as the array, with a cable terminus directly beneath the pad.

Instead of being immediately hurled at the station, ordered cargo is introduced to a pending cargo list; cargo is manually sent and received at each pad using Transception Interlink consoles. These consoles wirelessly connect to pads and can specify not only which pad to send or receive from, but also which pad is received if the available pending cargo differs.

This flexibility is not entirely without a price; array utilization exerts a grid power draw from the pad and array, as well as a direct cost from the array APC's power cell. Cell overuse will cause the transception functionality of the antenna to shut down temporarily as a precaution, returning when the cell's power climbs a bit. (Also, don't stand on the pads while they operate if you value your extremities.)

### Not Done Stuff

Things that I haven't touched on yet (either adequately, or at all) and will need to be tackled before map activation if this is merged. Likely not all-inclusive.

* High priority: Crystal windows need to be harder to disassemble - that or a new, extra-reinforced version of crystal windows should be created. One possible step I thought of is using a soldering iron to undo/redo an edge sealant.

* High priority: Facemelting is a bit of a problem. Ghosts from facemelted people should have the person's un-facemelted name, and there should ideally be some way to medically restore faces.

* High priority: Blobs should be disabled on the map until they can take damage from the acid - breaches currently play massively into their favor, with an external blob nucleus being nigh-unstoppable.

* A control console for finer management of the transception array and better reporting on its state of operation, as well as the ability to damage and repair the array. I have a mockup of how the GUI would look; please feel free to ask for it, or for the layered transception array sprite file.

* New reinforced floor tile sprites. They need to look "industrial but smooth", ideally with a sense that rods were involved in the reinforcement, and should probably have white/standard/black and color-edge variants. These are a somewhat high priority, as I'd like to handle installing them into the map once a suitable design's been figured out.

* User interface needs to be tuned up a lot for siphon and transception, probably with tgui. It's functional, but crude. In both cases, the tuneup would likely consist of different visuals and potentially a different input handling system for resonator parameters and import target respectively.

* Acid resilience needs to be more granular than "it disappears after a brief window or it doesn't". I would envision the touch reaction from the acid applying a dissolution effect (similar to current), but instead of deleting the item at the end, it reduces its durability based on how corrosion resistant the object is (with very low durabilities starting to permit acid damage to inner layers or the user). To avoid clutter, durability datum could be only created when it's needed, and for performance, the touch reaction wouldn't be reapplied while a dissolution is in progress.

* Body part coverage for clothing is probably pertinent for total immersion situations; chest/hand/foot/head damage would be different zones, and suit slot items would likely be able to prevent decay for hand and foot slots (bitflags, maybe?) - edge cases like shorts or t-shirts are probably fine since you're usually going to be out and about in a suit if you can help it, but a "PARTIAL PROTECTION" variable or flag could perhaps be used.

* (Note on the prior: Acid-immune gear would continue to not decay, but regular station acid-EVA gear should have a high but finite durability.)

* Bespoke acid suits that protect a little better against the cold than the current chemsuits, with variant sprites; "general"/civilian (i.e. EVA), engineering, security. These suits should have a long but finite period of safe contact with acid.

* Johnbus currently has no dock; was thinking of having it dock at the south escape wing's outer airlock, but can't figure out how to do that.

* Low priority: Map should probably have some sort of environmental events. Mass radiation events could feasibly penetrate given the acid pit's somewhat shallow depth; bespoke events could be rolling chunks of debris that hit and possibly breach the breach hull, and surface storm intensification that disrupts transception antenna operation and causes anyone outside on the station level to be blown around and chilled.

* Low priority: A new acid type could potentially exist for the ocean, less instantly lethal than fluoro but still something that kills you quickly without protection.

* Not sure if this can be a thing, but worth mentioning: Reinforced floors currently don't seem to have a bearing on explosions. If at all possible, I'd like to make it so most minor bombs (i.e. basic pipe bomb you assemble in like 60 seconds) hitting a reinforced station floor won't be able to singlehandedly cause a breach.

Small end notes: Pods are intended to be disabled for this map (Pilot trait and QM packages should be accounted for), but could be reactivated if they can be retooled to take acid damage. Diner and jazz lounge are both accounted for. Syndicate spacesuits are immune to acid, so that's taken care of for the time being.

## Why's this needed? <!-- Describe why you think this should be added to the game. -->

The philosophy behind the creation of this map was to offer a considerably different experience from other rotation maps, with a specific focus in mind.

Usually, going outside isn't all that scary unless you're off Z-level; Nadir's surroundings impose a continual threat, providing the feeling of being stuck in a tin can that's the only thing between you and death, something other stations understandably don't have a significant amount of. The new custom features provide something of an incentive to choose the map despite its dangers, which nicely parallels the in-universe motivation for the extraction site's creation.

Also, cool hexagon.

## Changelog

None (does not introduce map to rotation).

Momoberry commented:
Really exciting premise, even if it still needs polish, I'd love to try it as an open test or something on live.

ThatFiveGuys commented:
I like but think engine needs a lot of work, though for testing not a big deal.

SirSnek commented:
Real cool but there are some things I'd like to add. 

Interdepartmental cooperation is cool but sadly it can often falls flat becuase noone wants to work with another persoon. I don't think with the current systems in place sci will have a reason to ever work with engineering to operate the siphon and to fix this issue we'd either have to make a matsci rework that lets everyone get something out of matsci or add several useful elements to science that are based on matsci (and yes I know infusions are a thing but they're only ever used for gimmicks or chem duplication).

I have played engineer like 4 times so I can't talk much about the engine from a gameplay perspective but I do think a neat event that could occur if you really overcharge and abuse the engine would be a very large EMP or temporary grid shutdown, more variety than just a big boom.

The array will likely make qm life somewhat more interesting and that's good. Maybe it could be added as a thing you can construct on other maps with more cons?

Now onto the unfinished elements!
-A 2 second action bar for all of the steps should be a good enough solution to the plasmaglass window problem, not perfect mind you but it doesn't make them laughably easy to destroy.
-Something similar to that happened with borgs that got put into another body but the issue was fixed, could check that code for a solution.
-IMO the best course of action would be to make an existing upgrade or set of upgrades also give you acid protection, it'd make it more than "Spawn near wall, attack" and instead make the blob think more about where to spawn and how the crew could use it against them by letting acid into the room.
-The array shouldn't be something you can delete by using enough force as it's a cornerstone of a whole job. Slapping a board somewhere in tech storage works fine but it doesn't take much to go inside and take it, a better solution would be to make it somewhat resistant to explosives so that you need to take a risk when you want to destroy it but make it time consuming/costly to repair so that the people you fuck over will have to put in the same or slightly more effort to fix it. They let the crazy guy with a stick hit their precious technology afterall, it's on them.
-No comments on UI or spriting
-A robust system for item and armor durability would be amazing. Having your items just poof out of existance isn't fun and most of the gear that fully resists it is rare as frick.
-Body part based protection as a system is a bit controversial but I am for it. Scaling damage to the amount of exposed skin would be appreciated as it'd mirror getting a bunch of clothes on for higher thermal resistance to take less damage in space.
-Sounds good but I'd like to propose a special suit that only the CE gets, it gets charged with power and having a charged suit stops the durability drain from being in acid but the power is always going down, more if you're covered in acid. The suit can also be EMP'd to shortcircuit it and lower the durability+power by a ton.
-iirc the bus checks for the johnbus shuttle area and then spawns there when it's supposed to so you might need to add a john bus console to that lower dock but for all I know it could have some wacky code that'll make it harder. I might investigate more later.
-That or existing events could be alterted slightly to fit nadir more. Maybe ion storms would mess with the arrays and let QMs get special goodies.
-Capping the burn damage at 30 might be good enough. You still die and get horribly disfigured but slightly slower.
-Huh, never knew. Making it take less damage from explosions will encourage engineers to actually make them but if the stat is too high might be op and will need to be made into a special nadir only floor.

Kubius commented:
At least in personal server tests (biggest one did at least reach two digit pop), I've found that cooperative operation of the siphon worked nicely; researchers fiddle about with figuring out ideal setups for the more high-requirement extraction targets, engineers handle operating and constructing the resonators and picking out materials for catalytic rod manufacture.

On the topic of the array, absolutely concur on it only being damageable and not outright destructible. I don't think it's feasible to have it cohabitate with other cargo systems on other maps as a constructable; it overrides regular cargo reception entirely, and is (in-universe) a bespoke prototype not ready for standard production.

Acid protection wise, I absolutely agree on the topic of energy-protected acid suits as a default thing for Chief Engineer and a high-tier manufacturable item.

Random events wise, I suppose ion storms and other traditionally space-based hazards *might* be able to penetrate to an extent; Nadir is at a somewhat shallow depth sufficient for the transception array to operate, but I'm not sure how well they'd make it through the atmospheric conditions. (On the topic of those atmospheric conditions, restricting cargo transception to only be possible during clear to sparse cloud conditions on surface could be funky, potentially with acid hailstones or something falling down during particularly harsh storms.)

Momoberry commented:
I personally like acid, but for the cold/hot idea, I feel like you would need to change how temperature functions on crew in general, as it's incredibly forgiving to be too cold, and even in spots where something is too hot, half the time you won't be taking damage unless it's specifically atmospherics... really only space does cold damage, afaik. Bulkhead variant for shutters sounds incredibly thematic, and it would be nice having something different than firelocks as they can get pretty annoying to deal with. Honestly, i'd just like to give firelocks in general a manual actionbar to open them without a crowbar, even if it's fairly long, since you are completely trapped if the fire alarm is on the other side and that punishes newer players that don't prepare at the start of a round like a lot of veteran players.

As for the engineering issues, i don't think that's something that a map can really fix by itself. Engineering has had a significant issue for a while across all the maps, as there's just no incentive to waste half your round building and trying to fix things unless it's a heavy RP environment, and even then building tends to be slow and tedious, and a little confusing in terms of power sources and the like. The new engine would help a bit, which is why I was so excited for the map, but I think we need to throw engineering a bone and either make building less obnoxious when you have all the (proper tools), or condense their tools so they don't need to carry 13 items to cover their bases.

As an example, the CE alone would get a full set of tools, a deconstructor, an RCD, T-Ray, lamp manufacturer, the new apc pinpointer, the new material vacuum which can't even go in a backpack, glass/metal/wires/pipes, and anything else I've missed. It's not like they get any special storage pouches for these either, like security would for their tools. They just don't have the space to carry all of this with the item sizes and lack of storage. They also wear heavy ass, movement slowing suits that don't really have that many benefits beyond regular EVA suits, and they're usually dragging something behind them too. And then action bars for every construction they do, why would they bother. And then half of the construction isn't even documented in game, you have to go to the wiki.

TL;DR Give Engineering QOL, cause they have next to none compared to all the other departments. They have way too many tools to carry around and some of them should be condensed into combined tools, or even giving them a knockoff syndicate multitool without the welder. They are also way too slow due to what they have to carry around, and have no incentive to actually repair things outside of RP.

Momoberry commented:
> WRT cold: it's less of a primary environmental hazard and more of a piece of the environment. Getting melted is the main threat.
> 
> WRT engineering: Nadir was never expected to solve Engineering's issues by itself. The engineering QoL features you've mentioned sound great, but are something applicable to every map, not just Nadir, and I don't believe they're directly within the scope of this discussion or this PR.

Totally agree, just referencing that a map shouldn't be expected to solve the issues alone is all.

ZeWaka commented:
item.health exists for acid dissolving purposes btw, flockdrone absorbption uses it

ZeWaka commented on code/area.dm:
```diff
@@ -3657,7 +3657,9 @@ ABSTRACT_TYPE(/area/mining)
 	if(name == "Space" || src.name == "Ocean")			// override defaults for space
 		requires_power = 0
-		#ifdef UNDERWATER_MAP
+		#ifdef MAP_OVERRIDE_NADIR
```
this'll probably need to be an #if

ZeWaka commented on code/datums/gameticker.dm:
```diff
@@ -251,13 +251,22 @@ var/global/current_state = GAME_STATE_WORLD_INIT
 		participationRecorder.releaseHold()
 
+#ifdef MAP_OVERRIDE_NADIR
+	SPAWN(15000) // 25 minutes in; about the time catalytic generators' power peters out
```
what is 15000 that's a big number

Kubius replied:
fixing for this & counterpart

ZeWaka commented on code/datums/gameticker.dm:
```diff
@@ -251,13 +251,22 @@ var/global/current_state = GAME_STATE_WORLD_INIT
+	SPAWN(15000) // 25 minutes in; about the time catalytic generators' power peters out
+		for(var/obj/machinery/power/catalytic_generator/CG in machine_registry[MACHINES_POWER])
+			LAGCHECK(LAG_LOW)
+			if(CG?.gen_rate < 55000)
```
55000 whats

Kubius replied:
brain cells exploding (mine)

ZeWaka commented on code/map.dm:
```diff
@@ -30,6 +30,7 @@ var/global/list/mapNames = list(
 	"1 pamgoC" = 		list("id" = "PAMGOC", 		"settings" = "pamgoc", 			"playerPickable" = ASS_JAM),
 	"Kondaru" = 		list("id" = "KONDARU", 		"settings" = "kondaru", 		"playerPickable" = 1,		"MaxPlayersAllowed" = 80),
 	"Ozymandias" = 	list("id" = "OZYMANDIAS", "settings" = "ozymandias", 	"playerPickable" = 0, 	"MinPlayersAllowed" = 40),
+	"Nadir" = 		list("id" = "NADIR", 		"settings" = "nadir", 		"playerPickable" = 0),
```
would not want admins to be able to pick it

Kubius replied:
On the fence about that, honestly. While the map still needs some additional work to get it functioning precisely as intended, it's entirely playable and it would be useful to get some more information on how well not only the custom features but also the regular-station features operate.

That said: entirely alright with it being commented out if there's broad agreement that it should remain non-accessible until the feature set is built out.

ZeWaka commented on code/modules/fluids/fluid_turf.dm:
```diff
@@ -212,6 +213,14 @@
 				new /obj/overlay/tile_effect/cracks/spawner/pikaia(src)
 
+		if(spawningFlags & SPAWN_HOSTILE) //nothing good comes from acid-washed depths...
+			if (src.z == 5 && prob(1) && prob(4))
+				new /obj/critter/gunbot/drone/buzzdrone(src)
+			else if (src.z == 5 && prob(1) && prob(2))
+				new /obj/critter/gunbot/drone/cutterdrone(src)
+			else if (src.z == 5 && prob(1) && prob(1) && prob(25))
```
what is 5 i'm not good with numbers

ZeWaka replied:
also combine those probs lmao

ZeWaka commented on code/modules/siphon/siphon_minerals.dm:
```diff
@@ -0,0 +1,291 @@
+//should probably have highly specific parameters, maybe obtained through secrets?
+//idea in particular: the required shear is in the 96-105 range and chooses a different value every 60 or 90 sec
+//and the recipe is learned through a device that shows you when this value changes, so you need a rapidly adjustable resonator setup
+/*
```
why comment

Kubius replied:
Telecrystal synthesis would have significant implications in the Lore(TM), so it's not initially enabled; comments are a musing on the specifics of how it might happen.

ZeWaka commented on code/modules/power/catalytic_engine.dm:
```diff
@@ -0,0 +1,521 @@
+	var/gen_rate = 0
+	var/defer_updateicon = FALSE
+	var/output_tier = 0
+	var/running = 0
```
0 or false

Kubius replied:
Running should be true/false, yep. Remainder are numbers (which are 0 while inactive).

ZeWaka commented on code/modules/power/catalytic_engine.dm:
```diff
@@ -0,0 +1,521 @@
+			add_avail(gen_rate WATTS)
+
+		desc = "Current Output: [engineering_notation(gen_rate)]W"
+		if(gen_rate < 110000)
```
whats

Kubius replied:
on it

ZeWaka commented on code/modules/power/catalytic_engine.dm:
```diff
@@ -0,0 +1,521 @@
+			boutput(user, "<span class='alert'>You are too far away to do that.</span>")
+			return
+
+		if (get_dist(src,O) > 1)
```
shouldn't this be an interact check

Kubius replied:
probably

Kubius commented:
Changelog for 17d96a8:

- area.dm: Changed #ifdef MAP_OVERRIDE_NADIR to #if defined(MAP_OVERRIDE_NADIR)
- gameticker.dm: Changed delayed checks for engine status to use MINUTES.
- fluid_turf.dm: Changed hostile-only spawn conditions to use decimal probs and Z_LEVEL_MINING.
- catalytic_engine.dm: Removed vestigial "running" variable. Added WATTS to a gen_rate check. Changed mousedrop checks for catalytic rod units to correctly use in_interact_range.
- siphon_controls.dm: Added a check to validate whether an instance of resonator data still exists in the control console's list before attempting to reference its fields, fixing a runtime that cropped up in a new batch of testing.
- siphon_machinery.dm: Changed mousedrop checks for siphon unloading to correctly use in_interact_range.

Kubius commented:
Changelog for f1f0ca6:

- Added a small informational card placed on the siphon control bank that describes where the included information can be found, and bolded the glossary message in the Resonance Calibration Database to make it very clear that the glossary is important.
- Added escape subs. These may require additional layers of wall between them and the main escape corridors.
- Added a bot patrol route because apparently there was not one.
- Considerably improved intercom coverage around the station, particularly departmental intercoms.
- Reduced camera coverage in certain sections of Maintenance to add blind spots and facilitate creation of them, and added a camera to reach the public market APC.
- Dispersed additional AI status displays around the station.
- Fixed a tile misplacement in Research.
- Relocated Tanhony slightly. Unnamed pool monkey has taken a leave of absence.

Kubius commented:
@Garash2k Individual responses:

- There are not currently plans to introduce a PTL, but if some sort of vertical PTL were developed, it's not entirely off the table; a funky alternative might be retasking the transception array for power transmission.
- The cloner position is Probably Fine, but should it prove too awkward in practice, a reorganization of Medbay could occur down the line; there's enough space to experiment with alternate layouts.
- Elevator is at the moment primarily for escaping sub-levels in the event that you fall underground after a catastrophic breach. Not exclusively, but primarily.
- Siphon transmutation typically produces raw materials with a fairly regular structure; something so sophisticated as an artifact would be wildly difficult to synthesize. The primary draw of the Siphon for researchers is A) fun tinkering, B) a source of materials for experimental equipment, particularly the penultimate goal of starstone which permits all sorts of fun material infusions. It's not off the table that Research would have the opportunity to brave the acid pit with custom-tooled gear in order to acquire new artifacts.
- HoP does have a DWAINE computer, to integrate with the nearby scanner and printer for bureaucratic purposes.

nefarious6th commented:
very good overall. wont get into the acid mechanics or anything technical, just map design things:

- all other maps have the aux comm dish; this one only has the one transceiver setup. would you add another in somewhere?
- i think you watched me run around and miss cloning like three times when i was on the server; some windows on the east wall would help open it up and make it apparent where it is without having to rearrange medbay 
- double-check beepsky pathing and the beacons; my man is leaving high-priority doors to the engine core and science open!
- i respect the armory layout but i already kind of can see nobody is going to be able to break into it
- any way to squeeze an extra oven into the kitchen?
- double-check the wall cabinet in the genpop bathroom is a version that won't spawn with any surprise tools or anything
- i dont totally understand what transit to DOC and the syndie outpost would look like on this map without vehicles, but if i missed anything, please go ahead and clarify; from what i'm understanding, that again gets into an acid mechanics thing, and in that case, i dont have any brilliant ideas on how to go about that at this time and think it's probably best handled in the way that you outlined wanting to tackle it already.
- i know you mentioned on the server that the fluid spawn was unique so as not to generate any decorations on the turf and i'm really torn because it makes sense because it's an acid ocean, but the exterior also feels a little sparse. not high priority without vehicle exploration stuff fully implemented though.

- ptl notes? someone had a vertical version they were making. i can dig around for it and ping you if i find it. i don't know what would be required code-wise, but multiple stations could benefit from a vertical ptl at this point

think that's it.

Kubius commented:
Itemized replies:

- *Maybe.* If one is added, will probably be installed near the Armory.
- East wall is maintenance, but clarity of what is the cloning setup is something I can keep in mind during future tweaking.
- Can and will check beacons, yes.
- Could tune down the Armory defensives a bit. Breaking in is heavily predicated on access to an acid suit, which is a bit clunky; permitting delicate break-in so as not to release acid from underneath (bare plating and electrified grilles instead) might be a better approach.
- Extra oven, perhaps. Will take a peek.
- Genpop bathroom cabinet has towels.
- Syndie outpost wise, syndies have acid-resistant suits already; reaching it for purchases using an agent card is tricky, but emergency space suits should become at minimum barely adequate for a trip there and back if you make a beeline.
- Additional generated objects for ocean turfs are a probable addition down the line, though I'm not sure which shape they might take - will probably recruit help on this one.
- PTL: prior commentary plus shrug.

Kubius commented:
Update on bot beacon situation: Harner follows the planned patrol route precisely, not entering departments whatsoever. Beepsky completely disregards it for reasons I am not certain of.

Kubius commented:
Changelog for c8ca310:

- Further reduced starting charge of SMES (500k > 100k). As generation occurs for a considerable period of round start, effective start charge is still high.
- Changed arcade VR goggle cabinets to use arcade network goggles.
- Introduced a wall cabinet of spare harmonic systems calibrators to Research.
- Added an experimental local generator to Engineering, as an alternate tool for bootstrapping station operations if the assorted cells have already been taken.
- Added a CE access spawn to CE office because apparently it didn't have one.
- Added an additional door to QM/bridge lobby for symmetry, reorganizing surroundings a bit.
- Removed two cameras from the extraction nexus control room, as they were redundant.
- Introduced a Rockbox to the extraction nexus siphon room, and added an additional television camera.
- Added forcefields to the arrival shuttle.

Siphon mechanics section:

- Added two new high-shear siphon extraction targets: gold, and biomatter (blob chunks).
- Added support for unloading the siphon's materials reservoir into closets.
- Added reporting in the siphon systems control for effective extraction units per cycle, automatically summed up from active resonator configuration.

ZeWaka commented:
> Additional generated objects for ocean turfs are a probable addition down the line, though I'm not sure which shape they might take:
* Literal geothermal 'steam' acid vents?
* Acid monster critter dens
* pits to da depths
* giant bubble pillars caused by godly Alka-Seltzer tablet 
* rogue pirate acid harvesters
* hippie rave dens full of 'acid'
* graveyard full of people who have died in acid in other video games
* a vat of acid... inside an acid ocean
* assorted skeletons - but the skeletons are angry
* space mafia pushing people outside airlocks into the acid ocean
* pH test strips

Kubius commented:
Test failure for 3f0dc87 likely due to github api issues.

ZeWaka commented:
!merge_upstream

Kubius commented:
The map is fully operational as is, but certain features that would be ideal to have before rotation addition (most notably the changes to acid resilience) are systemic in nature, and my ability to trial the map is somewhat limited.

My request is that the map be integrated on a non-rotation basis in its current state with the intent of further testing and iteration taking place before a move to rotation, with the following benefits in mind:

- Ability for trial runs of the map to take place with secret submodule features enabled; some things that come to mind are ranching to see if the ranch is adequately designed, and send_to_brazil for its interaction with cargo transception.
- Ability to have larger-scale tests of the map, particularly with aforementioned features enabled; most tests I've conducted have had only a handful of respondents, with only one round reaching double-digit populace. I am not entirely confident in the current layout of the map, but I have a few ideas for iteration, and which direction I go should depend on collected feedback.
- More straightforward testing and integration on global mechanics that also affect Nadir, such as acid dissolution and crystal window reinforcement; anyone could develop a solution on a branch of main code and test it against Nadir easily, instead of having to continually combine their prototype code into a copy of the Nadir branch and continually ensure it's up to date with general main code changes.

Kubius commented:
Changes in e2d9e43:
- Corrected a goof in manufacturer blueprint defines after merge from master.
- Fixed some broken item paths for in-map instances.
- Changed some of the map's complement of energy-shield generators to door-shield generators.

Kubius commented:
Changes in 7d86fb8:

- Set the now-included door-shield generators to use power level 2 by default (atmosphere/liquid).
- Introduced additional shutters and blinds to various areas around the station, as well as doing a few fine adjustments to window complement (mainly reductions).

Kubius commented:
2df16de introduces some kinda-plant-kinda-rock doodads for sea and trench generation. There are initially two variants (glowing spires and breathing cairns called sinkspire and bitelung respectively), but I may add more; they're all diggable with their own table of possible dropped items.

Kubius commented:
e2e8640 replaces the vast majority of the station's reinforced floors with regular tiling, as well as adjusting a few other things like cable coverage, window electrification, random room setups and shield generator positions.

Kubius commented:
Main changes in 6f7164f:
- Removed some problematic peststart locations.
- Adjusted Security slightly; fewer windows, security weapons vendor in the secure overwatch room. Possibly more to come if this version of the map survives.
- Adjusted the eastern section of Medbay, eating into maintenance a bit; Genetics now has enough room to turn around in, and the treatment room has received enough space to feasibly serve as an area for short-term care as well as long-term.
- Medbay in general has had its window complement reduced slightly.

Kubius commented:
64f14ce consists primarily (but not exclusively) of a significant overhaul to Security, adding an interrogation room and hopefully providing an overall better "flow". Some minor alterations to these changes to fix goofs are happening over subsequent commits.

Kubius commented:
6d1882e introduces significantly more door-linked forcefields for breach containment (potentially to be changed for a different breach containment system or removed if breaches are made harder?), improves AI camera coverage in Security, adds additional ATMs and starts in on a QM modification.

Kubius commented:
3a4c733 improves the error code reporting from transception pads; the readout in the interlink console will now tell you whether the failure is due to a lack of connection, no array APC power or an array APC power warning. Cost directly charged to the APC has also been reduced from 150 to 80.

Kubius commented:
139b986 [bork fixed by 734949a] changes the behavior of aqua tenebrae a bit.

- If worn helmet and suit have the IMMUNE_TO_ACID flag, the acid's fluid pools will always not deal damage.
- Viscosity has been slightly increased to a value of 0.25 in the hope of reducing the rate at which floods will travel.
- Damage is now inflicted in rounded steps, so trivial quantities of damage will simply not happen at all instead.

Kubius commented:
f6df0cc changes the location in aqua tenebrae damage calculation where chemprot is applied, moving it to the damage itself; this makes chemprot act as a vastly more consistent protective factor.

Kubius commented:
Weird thing happening: Secret submodule placeholder(?) is now deciding to merge conflict periodically. Merging from master and then doing "git checkout HEAD +secret" will allow the merge to complete normally.

Kubius commented:
Changes in [d79e75c](https://github.com/goonstation/goonstation/pull/7730/commits/d79e75ce2aa40495aed643058c897b0dc511b677):

- Fixed a bug where players would gasp if protected from aqua tenebrae fluid by chemprot, but had enough of it to take no damage.
- Fixed a path omission in transception pad code that prevented certain types of artifact from being sold.
- Reduced the required extraction ticks for many higher-end siphon material targets.
- Removed an extra monkeyplant in Medbay.
- Corrected an incorrectly oriented arrow in the northeast navigation floor guide set.
- Added a mixer to the kitchen.
- Added a custom suicide for the Harmonic Siphon.

zjdtmkhzt commented on code/WorkInProgress/nadir_antenna.dm:
```diff
@@ -0,0 +1,688 @@
+							if(!sold_to_trader)
+								shippingmarket.sell_crate(thing2send)
+
+						else if(istype(thing2send,/obj/artifact))
+							shippingmarket.sell_artifact(thing2send)
+
+						else //how even
```
I don't think this works? Not all artifacts are /obj/artifact, you'd need to check for the artifact var on /obj or whatever it was, check the sell_artifact proc maybe

Kubius replied:
That sounds like probably a good thing to do, will check into that in a bit.

Kubius replied:
Done in feb07ad

Kubius commented:
Updates in [f0ba2d8](https://github.com/goonstation/goonstation/pull/7730/commits/f0ba2d8972f10f516526791ef386a85fb6dd333e)

Siphon

* Altered the warning given when too much extraction charge has accumulated in the siphon in an attempt to increase clarity of involved mechanics.

Map - Notable Features

* Added a cargo teleporter and associated charger to the artifact lab.
* Introduced extra signage to AI area entries for clarity.
* Rearranged Security a bit, removing an unnecessary taser recharger rack and adding a personal computer.
* Introduced an exit button to the secure release foyer.
* Added a second pipe dispenser cart, in northwest maintenance.
* Reduced conveyor movement rate in Disposals.
* Added two more copies of the siphon resonator blueprints to the tables in the extraction nexus, as well as a backup set in the Research Director's quarters.
* Improved Engineering's hardware complement, including three breach repair carts, an interdictor crate and additional metal foam grenades on the storeroom table. Also swapped locations of gas extractor and general manufacturer.
* Added a welding fuel tank to maintenance by the west side of Research.

Map - Misc

* Rearranged heads' quarters in bridge a bit, primarily with the addition of windows (in the case of the HoP, windows replacing their west door).
* Rearranged Robotics so it doesn't blow up when lights do.
* Reduced internal reinforcement in several low-importance station sections. 
* Fixed up windows in the AI core.

flappybatpal commented on code/map.dm:
```diff
@@ -31,6 +31,7 @@ var/global/list/mapNames = list(
 	"1 pamgoC" = 		list("id" = "PAMGOC", 		"settings" = "pamgoc", 			"playerPickable" = FALSE),
 	"Kondaru" = 		list("id" = "KONDARU", 		"settings" = "kondaru", 		"playerPickable" = TRUE,		"MaxPlayersAllowed" = 80),
 	"Ozymandias" = 	list("id" = "OZYMANDIAS", "settings" = "ozymandias", 	"playerPickable" = FALSE, 	"MinPlayersAllowed" = 40),
+	"Nadir" = 		list("id" = "NADIR", 		"settings" = "nadir", 		"playerPickable" = FALSE),
```
Probably "MaxPlayersAllowed" = 80?

Kubius replied:
Sounds like a sensible enough cap from how the tests played out.

Kubius replied:
While I'm in there, should I enable player pickable so the map is available for rotation after merge, or do I stick to the current approach of the rotation activation being a separate action?

frawhst commented on code/obj/sealab_objects.dm:
```diff
@@ -172,6 +172,115 @@
 	icon_state = "tubesponge_small"
 	database_id = "sea_plant_tubesponge-small"
 
+//NADIR DOODADS (indev)
```
Ordinarily I would say 'uncomment or remove' for the commented code but if this is being worked on then it's fine

frawhst commented on code/obj/sealab_objects.dm:
```diff
@@ -172,6 +172,115 @@
+
+	bullet_act()
+		break_apart()
+		return
```
```suggestion
```

frawhst commented on code/obj/sealab_objects.dm:
```diff
@@ -172,6 +172,115 @@
+
+	blob_act()
+		break_apart()
+		return
```
```suggestion
```

Kubius replied:
yep 2

frawhst commented on code/obj/sealab_objects.dm:
```diff
@@ -172,6 +172,115 @@
+
+	meteorhit()
+		break_apart()
+		return
```
```suggestion
```

frawhst commented on code/procs/jobprocs.dm:
```diff
@@ -433,6 +433,8 @@
 			H.traitHolder.removeTrait("pilot")
 			H.traitHolder.removeTrait("sleepy")
 			H.traitHolder.removeTrait("puritan")
+		if (map_setting == "NADIR") //Nadir: pilot trait screws the pilot and adds sub when sub should not otherwise exist.
+			H.traitHolder.removeTrait("pilot")
```
Maybe send a message to people so they don't get confused? "You weren't able to fly your pod because it would have melted in this environment!" or something

frawhst commented on tgui/packages/tgui/interfaces/TrscArray.js:
```diff
@@ -0,0 +1,79 @@
+import { Window } from '../layouts';
```
i see TGUI i stop code reviewing

frawhst reviewed (commented):
Few things:
Icon diff bot straight up isn't working so honor rule that this doesn't revert a bunch of sprites accidentally
Remove the secret ref change as that will fuck things up

This is Big Scary Review but the large majority of things are just 1-token changes (`TRUE/FALSE` defines, unnecessary returns, stuff like that) or requests for clarification

home stretch yo

Kubius commented on code/WorkInProgress/nadir_antenna.dm:
```diff
@@ -0,0 +1,689 @@
+		use_power(ARRAY_TELECOST)
+		playsound(src.loc, "sound/effects/mag_forcewall.ogg", 50, 0)
+		flick("beam",src.telebeam)
+		SPAWN(0.1 SECONDS)
+			src.is_transceiving = FALSE
```
Cooldown between transceptions. Clarifying this with a define.

Kubius commented on code/WorkInProgress/nadir_antenna.dm:
```diff
@@ -0,0 +1,689 @@
+	circuit_type = /obj/item/circuitboard/transception
+	object_flags = CAN_REPROGRAM_ACCESS
+	frequency = FREQ_TRANSCEPTION_SYS
+	var/net_id
```
I don't believe there's an inherent net ID variable on this kind of computer (I did double check and it definitely seems to be the case unless /obj/machinery/computer is defined in multiple places for some reason), so it shouldn't be redundant.

Kubius commented on code/WorkInProgress/nadir_antenna.dm:
```diff
@@ -0,0 +1,689 @@
+					manifest["Array Link"] = signal.data["opstat"]
+					src.known_pads[device_netid] = manifest
+					src.queue_dialog_update = TRUE
+		return
```
doin'

Kubius commented on code/WorkInProgress/nadir_antenna.dm:
```diff
@@ -0,0 +1,689 @@
+	frequency = FREQ_TRANSCEPTION_SYS
+	var/net_id
+	var/last_ping = 0
+	var/temp = null
```
Artifact of a previous UI setup. Removing.

Kubius commented on code/WorkInProgress/nadir_antenna.dm:
```diff
@@ -0,0 +1,689 @@
+				yell.data["command"] = "send"
+			SPAWN(0.5 SECONDS)
+				src.post_signal(yell)
+		return
```
yep

Kubius commented on code/WorkInProgress/nadir_antenna.dm:
```diff
@@ -0,0 +1,689 @@
+
+	user.Browse(HTML, "window=transception_\ref[src];title=Transception Interlink;size=350x550;")
+	onclose(user, "transception_\ref[src]")
+	return
```
aye

Kubius commented on code/WorkInProgress/nadir_antenna.dm:
```diff
@@ -0,0 +1,689 @@
+
+	src.formatted_list = rollingtext
+	src.list_is_updated = TRUE
+	return
```
sure

Kubius commented on code/WorkInProgress/nadir_antenna.dm:
```diff
@@ -0,0 +1,689 @@
+				src.build_command(manifest["INT_TARGETID"])
+
+	src.add_fingerprint(usr)
+	return
```
commencing

Kubius commented on code/WorkInProgress/nadir_antenna.dm:
```diff
@@ -0,0 +1,689 @@
+						var/sigindex = signal.data["data"]
+						if(isnum_safe(sigindex))
+							src.attempt_transceive(sigindex)
+		return
```
affirmative

Kubius commented on code/WorkInProgress/nadir_antenna.dm:
```diff
@@ -0,0 +1,689 @@
+			src.failsafe_active = FALSE
+			src.UpdateIcon()
+			. = TRUE
+		return
```
acknowledged

Kubius commented on code/map.dm:
```diff
@@ -908,6 +909,52 @@ var/global/list/mapNames = list(
+	window_layer_full = COG2_WINDOW_LAYER
+	window_layer_north = GRILLE_LAYER+0.1
+	window_layer_south = FLY_LAYER+1
+	auto_windows = 1
```
very true

Kubius commented on code/modules/chemistry/Reagents-PoisonEtc.dm:
```diff
@@ -937,6 +937,154 @@ datum
+						M.emote("scream")
+						M.TakeDamage("All", 0, volume / 6, 0, DAMAGE_BURN)
+					boutput(M, "<span class='alert'>The blueish acidic substance stings[volume < 6 ? " you, but isn't concentrated enough to harm you" : null]!</span>")
+				return
```
aye

Kubius commented on code/modules/chemistry/Reagents-PoisonEtc.dm:
```diff
@@ -937,6 +937,154 @@ datum
+					if (method == TOUCH && volume >= 30)
+						if (ishuman(M))
+							var/mob/living/carbon/human/H = M
+							var/blocked = 0
```
flip the switch

Kubius commented on code/modules/chemistry/Reagents-PoisonEtc.dm:
```diff
@@ -937,6 +937,154 @@ datum
+												K.changeStatus("acid", 5 SECONDS, list("mob_owner" = M))
+										else
+											H.visible_message("<span class='alert'>The blueish acidic substance slides off \the [K] harmlessly.</span>", "<span class='alert'>Your [H.wear_mask] protects you from the acid!</span>")
+										blocked = 1
```
switch the flip

Kubius commented on code/datums/jobs.dm:
```diff
@@ -1728,6 +1728,9 @@ ABSTRACT_TYPE(/datum/job/civilian)
 #elif defined(MAP_OVERRIDE_OSHAN)
 	limit = 1
 	special_spawn_location = null
+#elif defined(MAP_OVERRIDE_NADIR)
+	limit = 1
+	special_spawn_location = null
```
There is; it's located to the southeast.

Kubius commented on code/modules/materials/Mat_Fabrication.dm:
```diff
@@ -1,6 +1,7 @@
 /obj/machinery/nanofab/refining
 	name = "Nano-fabricator (Refining)"
 	blueprints = list(/datum/matfab_recipe/coilsmall,
+	/datum/matfab_recipe/catarod,
```
It actually was at one point, and that was the intent - I'm not sure why it isn't that way right now. Will correct promptly.

Kubius commented on code/modules/chemistry/Reagents-PoisonEtc.dm:
```diff
@@ -937,6 +937,154 @@ datum
+											D.changeStatus("acid", 5 SECONDS, list("mob_owner" = M))
+									else
+										H.visible_message("<span class='alert>The blueish acidic substance slides off \the [D] harmlessly.</span>", "<span class='alert'>Your [H.head] protects you from the acid!</span>")
+									blocked = 1
```
verily

Kubius commented on code/map.dm:
```diff
@@ -908,6 +909,52 @@ var/global/list/mapNames = list(
+
+	walls = /turf/simulated/wall/auto/supernorn
+	rwalls = /turf/simulated/wall/auto/reinforced/supernorn
+	auto_walls = 1
```
wallin'

Kubius commented on code/area.dm:
```diff
@@ -3725,7 +3728,7 @@ ABSTRACT_TYPE(/area/mining)
 //Halloween is all about darkspace
 	if(name == "Space" || src.name == "Ocean")			// override defaults for space
 		requires_power = 0
-		#ifdef UNDERWATER_MAP
+		#if defined(UNDERWATER_MAP)
```
Artifact of a previous ocean lighting hack I have since cleaned up. Swapping it back.

Kubius commented on code/modules/power/catalytic_engine.dm:
```diff
@@ -0,0 +1,519 @@
+				light.set_brightness(0.6)
+				light.enable()
+
+		. = GetOverlayImage("mask")
+		if(.)
+			UpdateOverlays(.,"mask")
```
From what I recall (been a bit since I messed with that), the purpose is to ensure any mask overlays applied by other sources will go over the top of the power indicators, but the one reference I found to that (flock ping) appears to now create a dummy object instead so I'm going to assume it's no longer necessary.

Kubius commented on code/modules/power/catalytic_engine.dm:
```diff
@@ -0,0 +1,519 @@
+	var/obj/machinery/catalytic_rod_unit/cathode_unit
+
+	var/gen_rate = 0
+	var/defer_updateicon = FALSE
```
Artifact of a mini optimization thing I was considering but that proved unnecessary. Removing it and commenting the surroundings anyway.

Kubius commented on code/modules/power/catalytic_engine.dm:
```diff
@@ -0,0 +1,519 @@
+		output_tier = 0
+
+		if(!src.anode_unit || !src.cathode_unit)
+			//if(!src.defer_updateicon)
```
delet

Kubius commented on code/modules/power/catalytic_engine.dm:
```diff
@@ -0,0 +1,519 @@
+		qdel(src.ovr_door)
+		qdel(src.ovr_rod)
+		qdel(src.ovr_clamp)
+		if(src.contained_rod) qdel(src.contained_rod)
```
Huh. Neat

Kubius commented on code/modules/power/catalytic_engine.dm:
```diff
@@ -0,0 +1,519 @@
+				O.set_loc(src)
+				src.contained_rod = O
+				src.rod_post_install()
+				return
```
commenced

Kubius commented on code/modules/power/catalytic_engine.dm:
```diff
@@ -0,0 +1,519 @@
+	proc/use_rod(var/expend_type)
+		. = src.contained_rod.expend_rod(src.gentype)
+		src.indicator_update()
+		return
```
minus return'd

Kubius commented on code/modules/power/catalytic_engine.dm:
```diff
@@ -0,0 +1,519 @@
+			src.toggling = FALSE
+
+		src.oldmode = src.mode
+		return
```
bonked

Kubius commented on code/modules/power/catalytic_engine.dm:
```diff
@@ -0,0 +1,519 @@
+	inhand_image_icon = 'icons/mob/inhand/hand_tools.dmi'
+	icon_state = "roditem-100"
+	item_state = "rods"
+	rand_pos = 1
```
trueified

Kubius commented on code/modules/power/catalytic_engine.dm:
```diff
@@ -0,0 +1,519 @@
+
+			newObj.setupMaterial()
+			newObj.set_loc(getOutputLocation(owner))
+		return
```
unreturned

Kubius commented on code/modules/siphon/siphon_controls.dm:
```diff
@@ -0,0 +1,493 @@
+			src.pair_siphon()
+
+	proc/pair_siphon()
+		src.paired_core?.paired_lever = null
+		src.paired_core = locate(/obj/machinery/siphon/core, orange(6,src))
```
Yep. Ensures that any current paired core that might exist is no longer referencing the lever.

Kubius commented on code/modules/power/catalytic_engine.dm:
```diff
@@ -0,0 +1,519 @@
+		..()
+
+	build(amount, var/obj/machinery/nanofab/owner)
+		for(var/i=0, i<amount, i++)
```
Giving it a whirl.

Kubius commented on code/modules/siphon/siphon_controls.dm:
```diff
@@ -0,0 +1,493 @@
+				src.attackby(I,user)
+				return
+
+			//Determine spot that was clicked
+			var/click_zone_override = null
+			switch(text2num(paramList["icon-x"]))
+				if(3 to 16) //Lever horizontal range
+					switch(text2num(paramList["icon-y"]))
+						if(11 to 28) //Lever vertical range
+							click_zone_override = CLICK_ZONE_LEVER
+				if(21 to 29) //Panel horizontal range
```
Same way the chem cabinets do regional click detection, and I don't know of another way. Definitely a bit cursed but it's worked in testing.

Kubius commented on code/modules/siphon/siphon_controls.dm:
```diff
@@ -0,0 +1,493 @@
+					if(signal.data["device"] == "SIPHON")
+						src.list_is_updated = FALSE
+						src.known_devices.Cut()
+		return
```
bonked

Kubius commented on code/modules/siphon/siphon_controls.dm:
```diff
@@ -0,0 +1,493 @@
+			yell.data += command_list
+			SPAWN(0.5 SECONDS)
+				src.post_signal(yell)
+		return
```
doin

Kubius commented on code/modules/siphon/siphon_controls.dm:
```diff
@@ -0,0 +1,493 @@
+	var/rollingtext = "" //list of entries for not siphon
+	var/intensity_sum = 0 //sum of resonator intensities (used to display effective extraction units per cycle, aka machine tick)
+
+	if(!length(src.known_devices))
+		mainlist = "<h2>NO CONNECTION TO DEVICES</h2><br>"
+		mainlist += "Displaying Default Message<br><br>"
+		mainlist += "Welcome to HARMONIC SIPHON CONTROL<br><br>"
```
Restructuring into a single bracketed formatted string.

Kubius commented on code/modules/siphon/siphon_controls.dm:
```diff
@@ -0,0 +1,493 @@
+
+/obj/machinery/computer/siphon_control/proc/build_formatted_list()
+	if(src.list_is_updated) return
+	var/mainlist = "" //held separately so the siphon can always start the list
+	var/rollingtext = "" //list of entries for not siphon
+	var/intensity_sum = 0 //sum of resonator intensities (used to display effective extraction units per cycle, aka machine tick)
```
Did a more extensive comment update.

Kubius commented on code/modules/siphon/siphon_controls.dm:
```diff
@@ -0,0 +1,493 @@
+				src.build_command(manifest["INT_TARGETID"],commanderino)
+
+	src.add_fingerprint(usr)
+	//src.updateUsrDialog() //not sure if this is needed given updates are prompted by receipt of update from a resonator
```
Seems to have been updating fine in live testing so I'll opt to just remove it for now.

Kubius commented on code/modules/siphon/siphon_controls.dm:
```diff
@@ -0,0 +1,493 @@
+
+	src.add_fingerprint(usr)
+	//src.updateUsrDialog() //not sure if this is needed given updates are prompted by receipt of update from a resonator
+	return
```
Bonked.

Kubius commented on code/modules/siphon/siphon_controls.dm:
```diff
@@ -0,0 +1,493 @@
+	///textified list of all indexed (non-hidden) minerals
+	var/mineral_list = null
+
+	New()
+		..()
```
Bonked.

Kubius commented on code/modules/siphon/siphon_controls.dm:
```diff
@@ -0,0 +1,493 @@
+	mainlist += rollingtext
+	src.formatted_list = mainlist
+	src.list_is_updated = TRUE
+	return
```
Bonked.

Kubius commented on code/obj/item/clothing/suits.dm:
```diff
@@ -1346,9 +1352,14 @@
 
 /obj/item/clothing/suit/space/industrial
+#ifdef MAP_OVERRIDE_NADIR
+	desc = "Armored, immersion-tight suit. Protects from a wide gamut of environmental hazards, including radiation and explosions."
```
"Immersion-tight" implies a diving suit. Could roll this into the #ifdef UNDERWATER_MAP?

frawhst replied:
Hmmm yeah that's fair. No issue then

Kubius commented on code/obj/item/device/nanoloom.dm:
```diff
@@ -0,0 +1,161 @@
+			user.put_in_hand_or_drop(loom_cart)
+			src.loom_cart = null
+			UpdateIcon()
+		return
```
Done did.

Kubius commented on code/modules/siphon/siphon_controls.dm:
```diff
@@ -0,0 +1,493 @@
+	if(..())
+		return
+
+	//var/subaction = (href_list["subaction"] ? href_list["subaction"] : null)
```
Artifact of when I was considering more than just calibration from the console. Probably better to remove it for cleanliness, even if I do end up adding something else later on.

Kubius commented on code/obj/item/device/nanoloom.dm:
```diff
@@ -0,0 +1,161 @@
+	icon = 'icons/obj/items/device.dmi'
+	icon_state = "nanoloom-active"
+	var/mob/living/user
+	var/obj/item/device/nanoloom/N
```
N for nanoloom probably not the clearest in context. Swapped to "loom".

Kubius commented on code/obj/item/device/nanoloom.dm:
```diff
@@ -0,0 +1,161 @@
+	icon_state = "nanoloom-cart"
+	w_class = W_CLASS_SMALL
+	inventory_counter_enabled = 1
+	rand_pos = 1
```
Yep.

Kubius commented on code/obj/item/device/nanoloom.dm:
```diff
@@ -0,0 +1,161 @@
+	icon = 'icons/obj/items/device.dmi'
+	icon_state = "nanoloom-cart"
+	w_class = W_CLASS_SMALL
+	inventory_counter_enabled = 1
```
Swabonked.

Kubius commented on code/obj/item/paper.dm:
```diff
@@ -1622,6 +1622,17 @@ exposed to overconfident outbursts on the part of individuals unqualifed to embo
+
+	New()
+		..()
+		info = "<html><body><style>img {width: 100%; height: auto;}></style><img src='[resource("images/pocket_guides/siphonguide.png")]'></body></html>"
```
Resource proc.

Kubius commented on code/WorkInProgress/nadir_antenna.dm:
```diff
@@ -0,0 +1,689 @@
+		return
+
+	proc/try_pad_ping()
+		if( (last_ping && ((last_ping + 10) >= world.time) ) || !src.net_id)
+			return 1
```
Sensible change. Did it.

Kubius commented on code/WorkInProgress/nadir_antenna.dm:
```diff
@@ -0,0 +1,689 @@
+	proc/telefrag(var/mob/living/carbon/human/M)
+		var/dethflavor = pick("suddenly vanishes","tears off in the teleport stream","disappears in a flash","violently disintegrates")
+
+		switch(rand(1,4))
```
The possibility of whooshing on a limb if you're already missing it is reasonably desirable to discourage use of the transception pad as a torture chamber; to avoid the weird behavior when rolling a limb that's already missing, I'm adding a check that makes the stun and scream only go off if you actually got a limb ripped off.

Kubius commented on code/modules/power/catalytic_engine.dm:
```diff
@@ -0,0 +1,519 @@
+	icon = 'icons/obj/machines/catalysis.dmi'
+	icon_state = "core"
+	anchored = 1
+	density = 1
```
I really dislike the feel of making this a TRUE, but I'll do it if you insist.

frawhst replied:
Yeah it does feel a little weird. I'm not insisting on it

virvatuli-gh reviewed (commented):
The siphon sound filesizes don't need to be as large as they are, nor do they really need to be in stereo, but it's not anything that will hold up the PR and I can just work on them a bit later. Sound fine otherwise. Please run the soundcache script so they're added properly to it though.

virvatuli-gh reviewed (changes requested):
The siphon sound filesizes don't need to be as large as they are, nor do they really need to be in stereo, but it's not anything that will hold up the PR and I can just work on them a bit later. Sound fine otherwise. Please run the soundcache script so they're added properly to it though.

Kubius commented:
Taking care of those oggs myself, since I have access to the raw audio. PR will update with them and the soundcache update soon.

nefarious6th commented:
Congrats on merge!
